docs/examples/nextjs-setup.md

```diff
@@ -0,0 +1,148 @@
+---
+id: nextjs-setup
+title: Nextjs Setup
+---
+
+## Set up your collection
+
+```typescript
+import { Collection } from 'datx';
+
+export default class AppStore extends Collection {
+  public static types = [Person];
+}
+```
+
+## _app page configuration
+
+There are couple of steps that you're gonna have to do in order to properly setup datx withyour next.js app.
+
+* You have to check wether you app is being served server side (with isServer() util) so your new collection instance can be properly return/created.
+* You will extend the default page context with your generated store instance.
+* Pass your store to the [Provider](https://github.com/mobxjs/mobx-react#provider-and-inject) component.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--TypeScript-->
+
+```tsx
+import App from 'next/app';
+import { Provider } from 'mobx-react';
+
+
+import { AppStore } from '../store/AppStore';
+import { isServer } from '../utils/isServer';
+
+
+function generateStore(snapshot?: any): AppStore {
+  if (isServer()) {
+    return new AppStore(snapshot);
+  }
+
+  if (!global['appStore']) {
+    global['appStore'] = new AppStore(snapshot);
+  }
+
+  return global['appStore'];
+}
+
+
+class MyApp extends App {
```

Comment:
I'm afraid this is not an optimal way of initializing the store. 
Read the comment here https://nextjs.org/docs/advanced-features/custom-app
It means we can't bailout from server-side rendering.

I made an example on gist https://gist.github.com/isBatak/883b5ae602146848c04f8544a33c368b to better explain the idea. Note the ssr boolean prop on withDatx, useStore and useObserver hooks.

Reply:
@DarkoKukovec could you check it also...
useObserver hook is explained here https://mobx-react.js.org/observer-hook

Reply:
Also, this is important https://mobx-react.js.org/recipes-ssr

Reply:
Looks good to me. Does that solve all the issues we discussed?

Reply:
If you asking about hydration problems then no :D, that problem still exists. 
In this example, we are finding all of the resources form the collection, but that might be incorrect because there could be more resources depending on what we fetched before this page.

The ideal solution would be something like this
```
import React from 'react';
import { useObserver } from 'mobx-react';
import { withDatx } from '../lib/with-datx';

const IndexPage = ({data}) => {  
  return useObserver(() => (
    <div>
      {data.map(...)}
    </div>
  )
}

IndexPage.getInitialProps = async ({ store }) => {
  const data = await store.fetchAll(...);

  return {
    data,
  }
}

export default withDatx(IndexPage)
```

Reply:
Ok, that's a bigger discussion then. Can we for now just add some message here with known issues or something, so the PR can be merged?

Reply:
At this moment I could remove next.js setup guide until we don't find a proper way to initialize the store?

Reply:
We can add the "COMING SOON" note and maybe point to gist?

Reply:
Sure, I will create a note and add a link to your gist for a current reference.